pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -243,6 +295,47 @@ protected GenericRow doUpdateRecord(GenericRow record, RecordInfo recordInfo) {
     }
   }
 
+  protected long loadWatermark() {
+    File watermarkFile = getWatermarkFile();
+    if (watermarkFile.exists()) {
```

Comment:
NIT: code would be much easier to read if it is structured as
if (!wmf.exist()) {
   return Long.MIN_VALUE;
} 
....

Reply:
@chenboat What if there is exception during load the watermark? 
we still want to return `Long.MIN_VALUE`, which is same as the case watermarkFile doesn't exist? 
Because of the above reason i check existence first and return loaded value.
if not exists or exception happens during try-catch, i return LONG.MIN_VALUE